lib/buffer.js

```diff
@@ -54,17 +54,19 @@ function alignPool() {
   }
 }
 
+const bufferDeprecationWarning =
+    deprecate(() => {}, 'Buffer constructor is deprecated. ' +
+      'Use Buffer.from, Buffer.allocUnsafe or Buffer.alloc instead.');
+
 /**
- * The Buffer() construtor is "soft deprecated" ... that is, it is deprecated
- * in the documentation and should not be used moving forward. Rather,
- * developers should use one of the three new factory APIs: Buffer.from(),
- * Buffer.allocUnsafe() or Buffer.alloc() based on their specific needs. There
- * is no hard deprecation because of the extent to which the Buffer constructor
- * is used in the ecosystem currently -- a hard deprecation would introduce too
- * much breakage at this time. It's not likely that the Buffer constructors
- * would ever actually be removed.
+ * The Buffer() construtor is deprecated ... that is, it should not be used
+ * moving forward. Rather, developers should use one of the three new factory
+ * APIs: Buffer.from(), Buffer.allocUnsafe() or Buffer.alloc() based on their
+ * specific needs. It's not likely that the Buffer constructors would ever
+ * actually be removed.
  **/
 function Buffer(arg, encodingOrOffset, length) {
+  bufferDeprecationWarning();
```

Comment:
@seishun https://github.com/addaleax/node/commit/ec09d43f9a5299f09c79e7c7af550b927fe31bc1 – Probably not the most performant way of doing anything, but it provides users with a subclassable `BufferClass` export that works exactly like the current `Buffer` constructor (all tests passing).

If the goal is to make `Buffer` itself subclassable, then one could work towards such a solution that would ultimately alias `Buffer` to what `BufferClass` is in my suggestion. That would break `Buffer()` without `new` but leave everything else intact.